File: src/ProjectR/content/scripts/spells/TicklingNeedles.lua
function GetTargetType()
    return Enemies
end

function GetName()
return "Tickling Needles"
end

function GetDescription()
    return "Throw magic infused needles at your enemies.\nWeak but cheap."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .24
end

function GetDelay()
  return .4
end

function GetSpellType()
  return Composite
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = (1.875 * aAD + .625 * aMD) - (.625 * dDEF + .3125 * dMR)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/TicklingNeedles.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class TicklingNeedles : SpellScriptBase
    {
        public override string Name { get { return "Tickling Needles"; } }
        public override string Description { get { return "Throw magic infused needles at your enemies.\nWeak but cheap."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Composite; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 24; } }
        public override double Delay { get { return .4; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (1.875 * aAD + .625 * aMD) - (.625 * dDEF + .3125 * dMR);
            DealDamage(damage);
        }
    }
}